Updated guidelines for building Pyre from source (#387)
facebook/pyre-check: documentation/website/docs/installation.md

This will compile the compiler from scratch and is likely going to take some time on your system.

> ### Building for OCaml changes

Reviewer: ```suggestion
### Building OCaml changes
```

$ make
$ make test
```
### Building for Pyre changes
In a virtualenv, install dependencies with `requirements.txt` and run python tests to make sure everything is set up correctly

```bash

> $ pip install -r requirements.txt

Reviewer: ```suggestion
$ cd /path/to/pyre-check
$ pip install -r requirements.txt
```

$ ./scripts/run-python-tests.sh
```
Since `pyre` is expected to run from the root directory using the `client/pyre.py` wrapper, we can add this directory to `PYTHONPATH` and subsequently assign it an alias to invoke the binary from anywhere

```bash
$ echo "export PYTHONPATH=/path/to/pyre-check" >> ~/.bashrc

> $ echo "alias pyre='python -m client.pyre'" >> ~/.bashrc

Reviewer: ```suggestion
$ echo "alias pyre='python -m client.pyre'" >> ~/.bashrc
$ source ~/.bashrc
```

```

> You should be able to open a new shell and run `pyre -h` now, confirming `pyre` was set-up correctly.

Reviewer: ```suggestion
You should be able run `pyre -h` now, confirming `pyre` was set-up correctly. Any changes made to the Pyre Python client code should be immediately observable the next time you invoke `pyre`
```

$ make
$ make test
```
### Testing changes to the Python Client
In a virtualenv, install dependencies with `requirements.txt` and run python tests to make sure everything is set up correctly

```bash
$ cd /path/to/pyre-check
$ pip install -r requirements.txt
$ ./scripts/run-python-tests.sh
```

> When installing and running `pyre` from PyPi, the entry point to the executable is actually `client/pyre.py`. To be able to run this file from anywhere, add the `pyre-check` directory to `PYTHONPATH` and subsequently assign `client.pyre` an alias to invoke the binary from anywhere

Reviewer: ```suggestion
When installing and running `pyre` from PyPi, the entry point to the executable is actually `client/pyre.py`. To be able to run this file from anywhere, add the `pyre-check` directory to `PYTHONPATH` and subsequently assign `pyre` and an alias for `client.pyre`.
```

```bash

> $ echo "alias pyre='PYTHONPATH=\"/path/to/pyre-check:\$PYTHONPATH\" python -m client.pyre'" >> ~/.bashrc

Reviewer: Just checking, you tested this latest variation on your own machine and it worked, right?

Author: I believe so, yes. `pyre` ran fine after that command. Moreover, my `.zshrc` file gets updated with the following line
![image](https://user-images.githubusercontent.com/17424778/109998429-38875a80-7d37-11eb-8056-b80e240c28b5.png)
after running the command, which i think is correct

$ make
$ make test
```
### Testing changes to the Python Client
In a virtualenv, install dependencies with `requirements.txt` and run python tests to make sure everything is set up correctly

```bash
$ cd /path/to/pyre-check
$ pip install -r requirements.txt
$ ./scripts/run-python-tests.sh
```
When installing and running `pyre` from PyPi, the entry point to the executable is actually `client/pyre.py`. To be able to run this file from anywhere, add the `pyre-check` directory to `PYTHONPATH` and subsequently assign `client.pyre` an alias to invoke the binary from anywhere

```bash
$ echo "export PYTHONPATH=/path/to/pyre-check" >> ~/.bashrc

> $ echo "alias pyre='python -m client.pyre'" >> ~/.bashrc

Reviewer 2: Just a suggestion: to avoid polluting the python path of the user, we could use:
```
echo "alias pyre='PYTHONPATH=/path/to/pyre-check python -m client.pyre'" >> ~/.bashrc
```

Reviewer: Ohhh I like this even better! Good suggestion.

On top of that change, we probably don't want to overwrite the exiting pythonpath, so maybe augment it using `:$PYTHONPATH`
```suggestion
$ echo "alias pyre='PYTHONPATH=/path/to/pyre-check python:$PYTHONPATH -m client.pyre'" >> ~/.bashrc
```

Reviewer 2: Good point. I think it should be:
```
$ echo "alias pyre='PYTHONPATH=\"/path/to/pyre-check:$PYTHONPATH\" python -m client.pyre'" >> ~/.bashrc
```
I think the quote is required if there is any space in the path.
